source/kraken/maintenance_worker.cpp

```diff
@@ -56,14 +56,116 @@ namespace bg = boost::gregorian;
 
 namespace navitia {
 
+void MaintenanceWorker::run() {
+    // try to load data from disk
+    load_data();
+
+    if (!is_data_loaded()) {
+        // The data is not loaded, so we wait for a reload message on the task queue
+        // that results in a successful data load.
+        // We may be disconnected from rabbitmq during this wait,
+        // and reconnection to rabbitmq is handled by this while(true) loop
+        while (true) {
+            try {
+                open_channel_to_rabbitmq();
+                create_task_queue();
+                // will return when a reload message on the task queue
+                // arrived and resulted in a successful data load
+                listen_to_task_queue_until_data_loaded();
+                // data is loaded, let's break from the while(true)  loop
+                break;
+            } catch (const std::runtime_error& ex) {
+                LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+                data_manager.get_data()->is_connected_to_rabbitmq = false;
+                channel_opened = false;
+                std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Just a suggestion...

```suggestion
                std::this_thread::sleep_for(std::chrono::seconds(conf.broker_sleeptime()));
```

Reply:
Now that I look carefully at where `conf.broker_sleeptime()` is used, its meaning is : "when I finished handling queues (i.e. checking if there is messages, and applying them if any) how long do I wait before looking at queues again".

Here, the duration means : "when connection to rabbitmq failed, how long do I wait before trying again to connect".

So I think I will introduce another parameter, rather than reuse `broker_sleeptime`

Reply:
I used `BROKER.reconnect_wait` as the new parameter